File: Using TDD with ASP.NET MVC/[C#]-Using TDD with ASP.NET MVC/C#/MvcContacts/Controllers/NotTDDController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Mvc;

using MvcContacts.Models;

namespace MvcContacts.Controllers {
    // <snippet22>
    public class NotTDDController : Controller {

        ContactEntities _db = new ContactEntities();

        public ActionResult Index() {
            ViewData["ControllerName"] = this.ToString();
            var dn = _db.Contacts;
            return View(dn);
        }

        public ActionResult Edit(int id) {
            Contact prd = _db.Contacts.FirstOrDefault(d => d.Id == id);
            return View(prd);
        }

        [HttpPost]
        public ActionResult Edit(int id, FormCollection collection) {
            Contact prd = _db.Contacts.FirstOrDefault(d => d.Id == id);
            UpdateModel(prd);
            _db.SaveChanges();
            return RedirectToAction("Index");
        }
    } 
    // </snippet22>
}

File: Using TDD with ASP.NET MVC/[VB]-Using TDD with ASP.NET MVC/VB/MvcContacts/Controllers/NotTDDController.vb
Namespace MvcContacts
    ' <snippet22>
    Public Class NotTDDController
        Inherits System.Web.Mvc.Controller

        Private _db As New ContactEntities()

        Public Function Index() As ActionResult
            ViewData("ControllerName") = Me.ToString()
            Dim dn = _db.Contacts
            Return View(dn)
        End Function

        Public Function Edit(ByVal id As Integer) As ActionResult
            Dim prd As Contact = _db.Contacts.FirstOrDefault(Function(d) d.Id = id)
            Return View(prd)
        End Function

        <HttpPost()>
        Public Function Edit(ByVal id As Integer, ByVal collection As FormCollection) As ActionResult
            Dim prd As Contact = _db.Contacts.FirstOrDefault(Function(d) d.Id = id)
            UpdateModel(prd)
            _db.SaveChanges()
            Return RedirectToAction("Index")
        End Function
    End Class
    ' </snippet22>
End Namespace